linuxprivchecker.sh

```diff
@@ -1,252 +1,251 @@
+## Permission is herby granted, free of charge, to any person obtaining a copy of this software and the
+## associated documentation files (the "Software"), to use, copy, modify, merge, publish, distribute, and/or
 ## sublicense copies of the Software, and to permit persons to whom the Software is furnished to do so, subject
 ## to the following conditions:
 ##
 ## The software must maintain original author attribution and may not be sold
 ## or incorporated into any commercial offering.
 ##
-## THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR ## IMPLIED, INCLUDING BUT NOT 
-## LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO 
-## EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER 
+## THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR ## IMPLIED, INCLUDING BUT NOT
+## LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+## EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER
 ## IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
 ## USE OR OTHER DEALINGS IN THE SOFTWARE.
 ###############################################################################################################
 
-### Useful functions
-
+# command paths
 PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games"
 
-TITLE_LINE=$(printf "%*s\n" "80" | tr ' ' "=")
-SECTION_LINE=$(printf "%*s\n" "80" | tr ' ' "-")
+# line formatting
+titleLINE=$(printf "%*s\n" "80" | tr ' ' "=");
+sectionLINE=$(printf "%*s\n" "40" | tr ' ' "-");
 
-function formatCommand(){
-	eval $1 | sed 's|^|    |'
+# title
+scriptTITLE(){
+echo ${titleLINE};
+echo "         LINUX PRIVILEGE ESCALATION CHECKER"
+echo "         https://github.com/linted/linuxprivchecker for more info..."
+echo ${titleLINE};
 }
 
-echo ${TITLE_LINE}
-echo "LINUX_PRIVILEGE ESCALATION CHECKER"
-echo ${TITLE_LINE}
-
-echo -e "\n[*] GETTING BASIC SYSTEM INFO...\n"
-
-echo "[+] Operating System"
-formatCommand "cat /etc/issue"
-
-echo -e "\n[+] Kernel"
-formatCommand "cat /proc/version"
-
-echo -e "\n[+] Hostname/FQDN"
-formatCommand "hostname -f"
+systemAREAtitle(){
+  echo ${sectionLINE};
+  echo -e "         $systemAREA";
+  echo ${sectionLINE};
+  echo
+  sleep .5s;
+}
 
-echo -ne "\n${SECTION_LINE}\n"
-echo -e "[*] GETTING NETWORKING INFO...\n"
+cmdRESPONSE(){
+	# run and format cmd
+  cmdRESULT=$(eval $1 2>/dev/null | sed 's|^|           |'; echo "${PIPESTATUS[0]}");
+
+	# check cmd status
+  if [ ${cmdRESULT:(-1)} -eq 0 ]; then
+    echo "[OK]     $systemNAME";
+    echo "${cmdRESULT%?}";
+    echo
+  else
+    echo "[FAILED] $systemNAME";
+    echo "${cmdRESULT%?}";
+    echo
+  fi
+  sleep .5s;
```

Comment:
@5aru @linted
The script has a 5 second delay for every title (line 55) command (line 72).
The delay allowed me to read things before the next cmd (personal preference). I picked this up from an OSCP troubleshooting script.
Glad you guys caught this but the speed was intentional.
Anyway by removing line 75 AND line 55, the script will be going as fast as before.